go/vt/sqlparser/ast_format.go

```diff
@@ -1199,6 +1199,9 @@ func (node *Subquery) Format(buf *TrackedBuffer) {
 
 // Format formats the node.
 func (node *DerivedTable) Format(buf *TrackedBuffer) {
+	if node.Lateral {
+		buf.WriteString(" lateral")
```

Comment:
are you sure about this space? I added an couple of examples to parse test, and they are failing:
```
	{
		input:  "select * from lateral (select 1) as x(user)",
		output: "select * from lateral (select 1 from dual) as x(`user`)",
	}, {
		input:  "select * from apa, lateral (select 1) as x(user)",
		output: "select * from apa, lateral (select 1 from dual) as x(`user`)",
	}
```

Output:
```
--- FAIL: TestValid (0.03s)
    --- FAIL: TestValid/select_*_from_lateral_(select_1)_as_x(user) (0.00s)
        parse_test.go:2318: Parsing failed.
            Expected/Got:
            select * from lateral (select 1 from dual) as x(`user`)
            select * from  lateral(select 1 from dual) as x(`user`)
    --- FAIL: TestValid/select_*_from_apa,_lateral_(select_1)_as_x(user) (0.00s)
        parse_test.go:2318: Parsing failed.
            Expected/Got:
            select * from apa, lateral (select 1 from dual) as x(`user`)
            select * from apa,  lateral(select 1 from dual) as x(`user`)
```

Reply:
Yeah, the space is on the wrong side of lateral! 😆

Reply:
My bad, I have fixed this.

Reply:
You should also add the tests that @systay ran into the parse_test file